src/fsharp/fsi/fsi.fs

```diff
@@ -2358,6 +2358,9 @@ type FsiEvaluationSession (fsiConfig: FsiEvaluationSessionHostConfig, argv:strin
     let defaultFSharpBinariesDir = System.AppDomain.CurrentDomain.BaseDirectory
     let currentDirectory = Directory.GetCurrentDirectory()
 #endif
+    // When used as part of FCS we cannot assume the current process is fsi.exe
+    // So we try to fallback to the default compiler dir.
+    let defaultFSharpBinariesDir = Internal.Utilities.FSharpEnvironment.BinFolderOfDefaultFSharpCompiler(None).Value
```

Comment:
Another possible solution would be to add the system directory (as proposed by this patch) at the end of https://github.com/fsharp/FSharp.Compiler.Service/blob/master/src/fsharp/build.fs#L2815